rt-forecast/update-death-from-cases.R

```diff
@@ -0,0 +1,72 @@
+# Packages -----------------------------------------------------------------
+library(EpiNow2, quietly = TRUE)
+library(data.table, quietly = TRUE)
+library(here, quietly = TRUE)
+library(future, quietly = TRUE)
+library(devtools, quietly = TRUE)
+library(lubridate, quietly = TRUE)
+library(ggplot2, quietly = TRUE)
+library(purrr, quietly = TRUE)
+
+# Set target date ---------------------------------------------------------
+target_date <- as.character(Sys.Date()) 
+
+# Get Observations --------------------------------------------------------
+deaths <- fread(here("data", "daily-incidence-deaths-Germany_Poland.csv"))
+cases <- fread(here("data", "daily-incidence-cases-Germany_Poland.csv"))
+deaths <- deaths[, secondary := value][, value := NULL]
+cases <- cases[, primary := value][, value := NULL]
+observations <- merge(cases, deaths, by = c("location", "location_name", "date"))
+observations <- observations[, .(region = as.character(location_name), date = as.Date(date), 
+                                 primary, secondary)]
+observations <- observations[date >= (max(date) - 8*7)][date <= target_date]
+setorder(observations, region, date)
+
+# Get case forecasts ------------------------------------------------------
+case_forecast <- suppressWarnings(
+  EpiNow2::get_regional_results(results_dir = here("rt-forecast", "data", "samples", "cases"),
+                                date = ymd(target_date),
+                                forecast = TRUE, samples = TRUE)$estimated_reported_cases$samples)
+
+# Forecast deaths from cases ----------------------------------------------
+# set up parallel options
+options(mc.cores = 4)
+plan("sequential")
+
+# load the prototype regional_secondary function
+source_gist("https://gist.github.com/seabbs/4dad3958ca8d83daca8f02b143d152e6")
+
+# run across Poland and Germany specifying options for estimate_secondary (EpiNow2)
+forecast <- regional_secondary(observations, case_forecast,
+                               delays = delay_opts(list(mean = 2.5, mean_sd = 0.2, 
+                                                        sd = 0.47, sd_sd = 0.1, max = 30)),
+                               secondary = secondary_opts(type = "incidence"),
+                               obs = obs_opts(scale = list(mean = 0.01, sd = 0.0025)),
+                               burn_in = as.integer(max(observations$date) - min(observations$date)) - 4*7,
+                               control = list(adapt_delta = 0.95, max_treedepth = 15))
+
+# Save results to disk ----------------------------------------------------
+
+samples_path <- here("rt-forecast", "data", "samples", "deaths-from-cases", target_date)
+summarised_path <- here("rt-forecast", "data", "summary", "deaths-from-cases", target_date)
+
+check_dir <- function(dir) {
+  if (!dir.exists(dir)) {
+    dir.create(dir, recursive = TRUE)
+  }
+  return(invisible(NULL))
```

Comment:
yes good point -  I was thinking we could do with a submissions tools thing somewhere as we reuse a lot of the same code again and again. I just like making sure that nothing is returned and it will definitely be silent. I don't think it is definitely needed. If not there it could return something (the first thing it finds above the }).